ports/boost-modular-build-helper/CMakeLists.txt

```
@@ -51,25 +51,46 @@ if(VCPKG_TARGET_ARCHITECTURE STREQUAL "x86" AND WIN32)
     list(APPEND B2_OPTIONS "asmflags=/safeseh")
 endif()
 
-if(MSVC)
-    set(USER_CONFIG_TOOLSET msvc)
-    if(MSVC_VERSION LESS 1900)
-        math(EXPR USER_CONFIG_TOOLSET_VERSION "${MSVC_VERSION} / 10 - 60")
+if(WIN32)
+    if(MSVC)
+        if (CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
+            if(CMAKE_CXX_COMPILER_FRONTEND_VARIANT STREQUAL "MSVC")
+                set(USER_CONFIG_TOOLSET clang-win)
+            else()
+                message(FATAL_ERROR "Unknown Clang frontend: ${CMAKE_CXX_COMPILER_FRONTEND_VARIANT}")
+            endif()

```

```suggestion
            set(USER_CONFIG_TOOLSET clang-win)
```

this can only be clang-cl in this branch.